### PR TITLE
Move nodeSelector, affinity and tolerations from templates to values

### PR DESCRIPTION
**Pull request by vvbogdanov87**
#44

### CODE REVIEW

File: charts/hdfs-datanode-k8s/values.yaml [hunk 1]
```diff
@@ -62,3 +62,8 @@ jsvcEnabled: true
 
 # Whether or not to expect namenodes in the HA setup.
 namenodeHAEnabled: true
+
+# Node labels and tolerations for pod assignment
+nodeSelector: {}
+tolerations: []
+affinity: {}
```
Comment (kimoonkim) [inline charts/hdfs-datanode-k8s/values.yaml]: Can we keep the current affinity code as a default here? i.e.
```
       nodeAffinity:	
          requiredDuringSchedulingIgnoredDuringExecution:	
            nodeSelectorTerms:	
              - matchExpressions:	
                - key: hdfs-datanode-exclude	
                  operator: DoesNotExist
```

File: charts/hdfs-journalnode-k8s/values.yaml [hunk 2]
```diff
@@ -50,3 +50,17 @@ kerberosConfigFileName: krb5.conf
...
+affinity:
+  podAntiAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+            - key: "app"
+              operator: In
+              values:
+              - hfds-journalnode
+        topologyKey: "kubernetes.io/hostname"
```
Comment (kimoonkim) [inline charts/hdfs-journalnode-k8s/values.yaml]: Hmm, there is some strange character at the end of the line?

**Comment by kimoonkim** [comment]
@vvbogdanov87, welcome to the project! And thanks for writing this PR. The approach sounds good to me. I am going to start looking at the change shortly.

File: charts/hdfs-namenode-k8s/values.yaml [hunk 3]
```diff
@@ -87,3 +84,17 @@ kerberosConfigFileName: krb5.conf
 # where HOST-NAME should match the cluster node
 # host name that each per-host hdfs principal is associated with.
 kerberosKeytabsSecret: hdfs-kerberos-keytabs
+
+# Node labels and tolerations for pod assignment
+nodeSelector: {}
```
Comment (kimoonkim) [inline charts/hdfs-namenode-k8s/values.yaml]: I wonder if we can add a comment here with the example of the pinning. i.e.
```
# It's possible to pin namenode pods to a set of nodes using nodeSelector.
# You can label your nodes like:	
#   $ kubectl label nodes YOUR-NODE hdfs-namenode-selector=hdfs-namenode
# Then specify nodeSelector like:	
# nodeSelector:	
#   hdfs-namenode-selector: hdfs-namenode
```

File: charts/hdfs-namenode-k8s/values.yaml [hunk 4]
```diff
@@ -39,9 +39,6 @@ zookeeperQuorum: zk-0.zk-svc.default.svc.cluster.local:2181,zk-1.zk-svc.default.
 # a standby namenode. Separated by the semicolon character.
 journalQuorum: hdfs-journalnode-0.hdfs-journalnode.default.svc.cluster.local:8485;hdfs-journalnode-1.hdfs-journalnode.default.svc.cluster.local:8485;hdfs-journalnode-2.hdfs-journalnode.default.svc.cluster.local:8485
 
-# Whether or not to enable pinning of namenode pods to labeled k8s cluster nodes.
```
Comment (kimoonkim) [inline charts/hdfs-namenode-k8s/values.yaml]: I found this flag is referred to by README.md. See https://github.com/vvbogdanov87/kubernetes-HDFS/blame/96720ecedf2d29d669b732059de3628c6782a9a2/charts/hdfs-namenode-k8s/README.md#L134

Can you please remove the section?

File: charts/hdfs-namenode-k8s/values.yaml (hunk 3 above)
Comment (vvbogdanov87) [reply charts/hdfs-namenode-k8s/values.yaml]: I believe that for pro users it should be obvious what namenodeSelector is. For others, I fixed namenode readme.

Comment (kimoonkim) [reply charts/hdfs-namenode-k8s/values.yaml]: Fair enough. Thanks!

**Comment by kimoonkim** [comment]
LGTM and merging soon. Thanks!

**Pull request merged**